remove gcc intrinsic and include-fixed include directories from analysis

These directories when added to the include lists by default cause some analysis action to fail.

For example by analyzing VIM (https://github.com/vim), we get the following errors:

```
./reports-vim/gui_gtk_f.c_0199d975c9e6b3f90bab69388afd723d.plist
./reports-vim/mark.c_7148c458eda4223ef34ea510d0199c9f.plist
./reports-vim/encoding.c_62589f05ad3f5b2abb93ee084903271c.plist
(venv)ednikru@md-mtas2:~/work/codechecker$ less ./vim.log 
(venv)ednikru@md-mtas2:~/work/codechecker$ cat vim.log 
[14:23] - Starting static analysis ...
[14:23] - [1/158] clangsa analyzed keyboard.c successfully.
[14:23] - [2/158] clangsa analyzed vterm.c successfully.
[ERROR] [14:24] - Analyzing 'dict.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/dict.c:14:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'list.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/list.c:14:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'crypt_zip.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/crypt_zip.c:13:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'charset.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/charset.c:10:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'osdef0.c' with clangsa failed.
[ERROR] [14:24] - clang-5.0: error: no such file or directory: '/home/ednikru/work/codechecker/vim/src/osdef0.c'
clang-5.0: error: no input files

[ERROR] [14:24] - Analyzing 'digraph.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/digraph.c:14:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'popupmnu.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/popupmnu.c:13:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'if_cscope.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/if_cscope.c:12:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'search.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/search.c:13:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'pty.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/pty.c:44:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'sha256.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/sha256.c:23:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'hardcopy.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/hardcopy.c:14:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'window.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/window.c:10:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'gui_gtk_x11.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/gui_gtk_x11.c:28:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'ex_cmds2.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/ex_cmds2.c:14:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'ops.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/ops.c:15:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'gui.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/gui.c:11:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'ex_getln.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/ex_getln.c:14:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'regexp.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/regexp.c:47:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'version.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/version.c:10:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'evalfunc.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/evalfunc.c:15:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'syntax.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/syntax.c:14:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'pathdef.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/auto/pathdef.c:4:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[14:24] - [26/158] clangsa analyzed mouse.c successfully.
[ERROR] [14:24] - Analyzing 'arabic.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/arabic.c:16:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'json.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/json.c:17:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'crypt.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/crypt.c:13:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'undo.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/undo.c:87:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'menu.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/menu.c:15:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'gui_gtk_f.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/gui_gtk_f.c:28:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'fold.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/fold.c:15:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'netbeans.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/netbeans.c:26:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'blowfish.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/blowfish.c:22:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'memline.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/memline.c:45:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'ex_eval.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/ex_eval.c:14:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'move.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/move.c:20:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.

[ERROR] [14:24] - Analyzing 'hashtab.c' with clangsa failed.
[ERROR] [14:24] - In file included from /home/ednikru/work/codechecker/vim/src/hashtab.c:30:
In file included from /home/ednikru/work/codechecker/vim/src/vim.h:2099:
In file included from /home/ednikru/work/codechecker/vim/src/spell.h:267:
In file included from /usr/include/wchar.h:884:
/usr/include/x86_64-linux-gnu/bits/wchar2.h:448:3: error: "Assumed value of MB_LEN_MAX wrong"
# error "Assumed value of MB_LEN_MAX wrong"
  ^
1 error generated.
```

--- a/libcodechecker/analyze/log_parser.py
+++ b/libcodechecker/analyze/log_parser.py
@@ -69,14 +69,48 @@ def parse_compiler_includes(lines):
             # /System/Library/Frameworks (framework directory)
             fpos = line.find("(framework directory)")
             if fpos == -1:
-                include_paths.append("-isystem " + line)
+                include_paths.append(line)
             else:
-                include_paths.append("-isystem " + line[0:fpos-1])
+                include_paths.append(line[0:fpos-1])
 
         if line.startswith(start_mark):
             do_append = True
 
     return include_paths
+
+
+def filter_compiler_includes(include_dirs):
+    """
+    Filter the list of compiler includes.
+    We want to elide GCC's include-fixed and instrinsic directory.
+    See docs/gcc_incompatibilities.md
+    """
+
+    def contains_intrinsic_headers(include_dir):
+        """
+        Returns True if the given directory contains at least one intrinsic
+        header.
+        """
+        if not os.path.exists(include_dir):
+            return False
+        for f in os.listdir(include_dir):
+            if f.endswith("intrin.h"):
+                return True
+        return False
+
+    result = []
+    for include_dir in include_dirs:
+        # Skip GCC's fixinclude dir
+        if os.path.basename(os.path.normpath(include_dir)) == "include-fixed":
+            continue
+        if contains_intrinsic_headers(include_dir):
+            continue
+        result.append(include_dir)
+    return result
+
+
+def prepend_isystem_and_normalize(include_dirs):
+    return ["-isystem " + os.path.normpath(idir) for idir in include_dirs]
 
 
 def parse_compiler_target(lines):
@@ -146,7 +180,8 @@ def get_compiler_includes(parseLogOptions, compiler, lang, compile_opts,
         dump_compiler_info(parseLogOptions.output_path,
                            compiler_includes_dump_file,
                            {compiler: err})
-    return parse_compiler_includes(err)
+    return prepend_isystem_and_normalize(
+        filter_compiler_includes(parse_compiler_includes(err)))
 
 
 def get_compiler_target(parseLogOptions, compiler):

